### ISSUE TITLE
Need own VCS-servers for testing the tool

### ISSUE THREAD

**Issue by fzzr-** [HaxeFoundation/haxelib#264]
I think we should not depend on Github or Bitbucket (or any other) in the testing the CVS-features of Haxelib tool.

We need our own independent VCS-servers for automated (unit-) tests for the Haxelib console tool.

All servers have to support http, https and ssh connections.

List of needed repos:
* Git without any auth
* Git __with__ login-pass auth
* Git __with__ cert/key auth
* Hg without any auth
* Hg __with__ login-pass auth
* Hg __with__ cert/key auth
* Svn without any auth
* Svn __with__ login-pass auth
* Svn __with__ cert/key auth
* Maybe [Bazaar](https://en.wikipedia.org/wiki/GNU_Bazaar) too

_Also we need more tests (unit + integration)._

**Comment by fzzr-** [comment]
Alternatively the servers can be launched on CI but it's not an easy quest.

**Comment by ibilon** [comment]
Agree, but wouldn't be a good idea to make a public server without auth,
and it'd be better for CI to have a fresh server too.

I took a look on how to create a git server, and it looks like it needs a new user,
travis can give sudo so could be possible, or maybe we can do another way that may be less secure but since it's just for testing.

**Comment by RealyUniqueName** [comment]
As a cheap solution we can go with directory-based repos. Both hg and git can use local directories e.g. to clone from. I doubt we need to support svn.
